libraries/chain/system_calls.cpp

```diff
@@ -191,29 +155,41 @@ struct block_setter
 THUNK_DEFINE( void, apply_block,
    (
       (const protocol::block&) block,
-      (boolean) check_passive_data,
-      (boolean) check_block_signature,
-      (boolean) check_transaction_signatures)
+      (bool) check_passive_data,
+      (bool) check_block_signature,
+      (bool) check_transaction_signatures)
    )
 {
    KOINOS_TODO( "Check previous block hash, height, timestamp, and specify allowed set of hashing algorithms" );
 
-   KOINOS_ASSERT( !context.is_in_user_code(), insufficient_privileges, "Calling privileged thunk from non-privileged code" );
+   KOINOS_ASSERT( !context.is_in_user_code(), insufficient_privileges, "calling privileged thunk from non-privileged code" );
 
    auto setter = block_setter( context, block );
-   block.active_data.unbox();
-
-   const multihash& tx_root = block.active_data->transaction_merkle_root;
-   size_t tx_count = block.transactions.size();
+   KOINOS_TODO( "Should this be a more specific exception?" );
```

Comment:
We need more of this type of data checking on everything else. If not as part of the conversion, we need to create an issue for it.